src/validation/logic.rs

```diff
@@ -53,16 +54,18 @@ impl Schema {
             redeem_schema,
             assign_schema,
             valency_schema,
-        ) = match (op.transition_type(), op.extension_type()) {
-            (None, None) => {
-                // Right now we do not have actions to implement; but later
-                // we may have embedded procedures which must be verified
-                // here
-                /*
-                if let Some(procedure) = self.genesis.abi.get(&GenesisAction::NoOp) {
-
+        ) = match (op, op.transition_type(), op.extension_type()) {
```

Comment:
We can drop the `unreachable!` branch and let the reader know the types are only included in specific op.